pkgs/native_assets_cli/lib/src/link_mode.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'api/asset.dart';
+import 'api/build_config.dart';
+
+/// The link mode for a [NativeCodeAsset].
+///
+/// Known linking modes:
+///
+/// * [StaticLinking]
+/// * [DynamicLoading]
+///   * [LookupInProcess]
+///   * [LookupInExecutable]
+///   * [DynamicLoadingBundled]
+///   * [DynamicLoadingSystem]
+///
+/// See the documentation on the above classes.
+sealed class LinkMode {
+  const LinkMode();
```

Comment:
Can this constructor be private since all sub classes are in the same file?

Reply:
Sure.